plugins/inputs/varnish_cache/varnish_cache.go

```diff
@@ -0,0 +1,321 @@
+			vValue float64
+			//iValue      uint64
+			vErr error
+		)
+
+		flag := data["flag"]
+
+		// parse value
+		if value, ok := data["value"]; ok {
+			if number, ok := value.(json.Number); ok {
+				if vValue, vErr = number.Float64(); vErr != nil {
+					vErr = fmt.Errorf("%s value float64 error: %s", vFieldName, vErr)
+				}
+				//Happy health probes TODO
+				if flag == "b" {
+					if _, vErr = strconv.ParseUint(number.String(), 10, 64); vErr != nil {
+						vErr = fmt.Errorf("%s value uint64 error: %s", vFieldName, vErr)
+					}
+				}
+			} else {
+				vErr = fmt.Errorf("%s value it not a float64", vFieldName)
+			}
+		}
+
+		if vErr != nil {
+			acc.AddError(vErr)
+			continue
+		}
+
+		measurement, field, tags := createMetric(vFieldName)
+		fields := make(map[string]interface{})
+		fields[field] = vValue
+		switch flag {
+		case "c", "a":
+			acc.AddCounter(measurement, fields, tags, timestamp)
+		case "g":
+			acc.AddGauge(measurement, fields, tags, timestamp)
+		default:
+			acc.AddGauge(measurement, fields, tags, timestamp)
+		}
+	}
+	return nil
+}
+
+// converts varnish metrics name into field and list of tags
+func createMetric(vName string) (measurement string, name string, tags map[string]string) {
+	measurement = extractMeasurement(vName)
+	var innerPart string
+	if strings.Count(vName, ".") > 1 {
+		innerPart = trimMeasurementPrefix(strings.ToLower(vName))
+		innerPart = innerPart[0:strings.LastIndex(innerPart, ".")]
+	}
+	name = trimMeasurementPrefix(strings.ToLower(vName))
+	if len(innerPart) > 0 {
+		name = strings.Replace(name, strings.ToLower(innerPart)+".", "", -1)
+	}
+	name = strings.Replace(name, ".", "_", -1)
+	tags = make(map[string]string)
+	if len(name) > 0 {
+		if isVBE := strings.HasPrefix(vName, "VBE."); isVBE {
+			if hits := regexBackendUUID.FindAllStringSubmatch(innerPart, -1); len(hits) > 0 && len(hits[0]) >= 3 {
+				tags["backend"] = cleanBackendName(hits[0][2])
+				tags["server"] = hits[0][1]
+			} else if hits := regexBackendParen.FindAllStringSubmatch(innerPart, -1); len(hits) > 0 && len(hits[0]) >= 3 {
+				tags["backend"] = cleanBackendName(hits[0][1])
+				tags["server"] = strings.Replace(hits[0][2], ",,", ":", 1)
+			}
+			if len(tags) == 0 {
+				tags["backend"] = cleanBackendName(innerPart)
+			}
+		}
+		if len(tags) == 0 && innerPart != "" {
+			tags["id"] = innerPart
+		}
+	}
+	return measurement, name, tags
+}
+
+func cleanBackendName(name string) string {
+	name = strings.Trim(name, ".")
+	for _, prefix := range []string{"boot.", "root:"} {
+		name = strings.TrimPrefix(name, prefix)
+	}
+	// reload_20191014_091124_78599.<name>
+	if strings.HasPrefix(name, "reload_") {
+		dot := strings.Index(name, ".")
+		if dot != -1 {
+			name = name[dot+1:]
+		}
+	}
+	return name
+}
+
+// Find the most recent 'VBE.reload_' prefix using string compare
+// 'VBE.reload_20210623_170621_31083'
+func findActiveReloadPrefix(countersJSON map[string]interface{}) string {
```

Comment:
This approach only work if you reload services, but if you load vcl manually with custom name doesn't valid

Reply:
I reworked this PR, now it is part of the varnish plugin, varnishadm tool is used to get active vcl.

Reply:
@ismaelpuerto is this concern resolved?